src/rust/bitbox02-rust/src/bb02_async.rs

```diff
@@ -44,6 +44,13 @@ impl<O> core::future::Future for AsyncOption<'_, O> {
     }
 }
 
+pub fn option_no_screen<O>(option: &RefCell<Option<O>>) -> AsyncOption<O> {
+    bitbox02::ui::screen_saver_disable();
+    let option_result = AsyncOption(option);
```

Comment:
this is not actually blocking here and the screen saver will be enabled immediately again.

you want to make  this function async

 ```Rust
pub async fn option_no_screensaver<O>(opt: &RefCell<Option<O>>) -> O {
// disable screensaver
let result = option(opt).await;
// enable screensaver
result
}
```

or similar.

Reply:
Thank you! :)